justbuild.it-WebApp/src/test/java/justbuild/it/web/app/service/OfferSearchingServiceTest.java

```diff
@@ -0,0 +1,144 @@
+import justbuild.it.web.app.repository.OfferFileRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class OfferSearchingServiceTest {
+    private OfferFileRepository offerFileRepositoryMock;
+    private OfferSearchingService offerSearchingService;
+
+    private static final Offer OFFER_1 = new Offer(1L, ServiceCategoryEnum.CONSTRUCTION, "Przykładowa oferta budowy", "Gdańsk", new User("Jan", "Kowalski", "KowalBud", "kowal@example.com", "111222333"), LocalDateTime.now().minusHours(1));
+    private static final Offer OFFER_2 = new Offer(2L, ServiceCategoryEnum.GARDEN, "Projektowanie ogrodu", "Szczecin", new User("Filip", "Nowak", "NowakGarden", "nowak@example.com", "123456789"), LocalDateTime.now().minusDays(1));
+    private static final Offer OFFER_3 = new Offer(3L, ServiceCategoryEnum.FINISHING_WORKS, "Prace wykończeniowe", "Warszawa", new User("Roman", "Romanowski", "RomanBud", "romanBud@example.com", "987654321"), LocalDateTime.now().minusDays(2));
+    private static final Offer OFFER_4 = new Offer(4L, ServiceCategoryEnum.EARTH_WORKS, "Prace ziemne", "Warszawa", new User("Tomasz", "Filipiak", "Filipiak&Son", "filipiak@example.com", "435657345"), LocalDateTime.now().minusDays(8));
+    private static final Offer OFFER_5 = new Offer(5L, ServiceCategoryEnum.ELECTRICITY, "Instalacje elektryczne", "Radom", new User("Tadeusz", "Wolski", "InstallWol", "install@example.com", "435657345"), LocalDateTime.now().minusDays(1));
+    private static final Offer OFFER_6 = new Offer(6L, ServiceCategoryEnum.INSTALLATION, "Kanalizacja", "Kraków", new User("Andrzej", "Redlin", "RedLine", "redline@example.com", "567897546"), LocalDateTime.now().minusDays(3));
+    private static final Offer OFFER_7 = new Offer(7L, ServiceCategoryEnum.GARDEN, "Aranżacja ogrodu", "Nowy Sącz", new User("Eustachy", "Konieczko", "Gardens", "gardens@example.com", "1232343212"), LocalDateTime.now().minusDays(6));
+
+    private static final List<Offer> GIVEN_TEST_LIST = List.of(OFFER_1, OFFER_2, OFFER_3, OFFER_4, OFFER_5, OFFER_6, OFFER_7);
+
+    @BeforeEach
+    public void setUpTest() {
+        offerFileRepositoryMock = mock(OfferFileRepository.class);
+        offerSearchingService = new OfferSearchingService(offerFileRepositoryMock);
+    }
+
+    @AfterEach
+    public void restoreSetUpTest() {
+        offerFileRepositoryMock = null;
+        offerSearchingService = null;
```

Comment:
hmmmm troche nie rozumie tego. Jesteś pewien, że to jest potrzebne?

Zobacz czy ta metoda jest w ogóle potrzebna. Wydaje mi się, że zmienne są automatycznie usuwane po zakończeniu każdego testu.